src/main/java/io/kaitai/struct/visualizer/VisualizerPanel.java

```diff
@@ -34,7 +39,17 @@
 
 public class VisualizerPanel extends JPanel {
     private static final String DEST_PACKAGE = "io.kaitai.struct.visualized";
-    private static final Pattern TOP_CLASS_NAME = Pattern.compile("public class (.*?) extends KaitaiStruct");
+    /**
+     * Regexp with 2 groups: class name and type parameters. Type parameters
+     * must be parsed with {@link #PARAMETER_NAME}.
+     */
+    private static final Pattern TOP_CLASS_NAME_AND_PARAMETERS = Pattern.compile(
+        "public class (.+?) extends KaitaiStruct.*" +
+        "public \\1\\(KaitaiStream _io, KaitaiStruct _parent, \\1 _root(.*?)\\)",
+        Pattern.DOTALL
+    );
```

Comment:
Actually, nowadays top class name should not be that hard to get. Probably we should expose [JavaMain.compileOneInput](https://github.com/kaitai-io/kaitai_struct_compiler/blob/master/jvm/src/main/scala/io/kaitai/struct/JavaMain.scala#L277) that will clearly return compilation result as Java object, and that object [contains reference to top class name](https://github.com/kaitai-io/kaitai_struct_compiler/blob/master/jvm/src/main/scala/io/kaitai/struct/JavaMain.scala#L291).